src/config.rs

```diff
@@ -98,10 +98,35 @@ pub struct DepembedsConfig {
 }
 
 /// Hyperparameters for Subword vocabs.
+#[derive(Clone, Copy, Debug)]
+pub struct SubwordVocabConfig<V> {
+    /// Vocabulary cutoff options.
+    ///
+    /// Vocabulary size cut-off.
+    pub vocab_cutoff: VocabCutoff,
+
+    /// Discard threshold.
+    ///
+    /// The discard threshold is used to compute the discard probability of
+    /// a token. E.g. with a threshold of 0.00001 tokens with approximately
+    /// that probability will never be discarded.
+    pub discard_threshold: f32,
+
+    /// Minimum n-gram length for subword units (inclusive).
+    pub min_n: u32,
+
+    /// Maximum n-gram length for subword units (inclusive).
+    pub max_n: u32,
+
+    /// Indexer specific parameters.
+    pub indexer: V,
+}
+
+/// Hyperparameters for Subword vocabs  that are going to be written into embeddings.
 #[derive(Clone, Copy, Debug, Serialize)]
 #[serde(rename = "SubwordVocab")]
 #[serde(tag = "type")]
-pub struct SubwordVocabConfig<V> {
+pub struct SubwordVocabWriteConfig<V> {
```

Comment:
I found out that, apparently, the TOML crate [doesn't seem to support newtype variants in enums](https://docs.rs/toml/0.5.5/src/toml/value.rs.html#826-837). It seems like a bug to me. So I separate the config that is used to build ```vocab``` and the config that is going to be serialized and written into ```Embeddings```

Reply:
I think we should not carbon copy such structs. Wouldn't it be possible to implement a custom serializer for `VocabCutoff`?